### DISCUSSION TITLE
Background onClick event

### DISCUSSION THREAD

**Discussion by noam-isaac** [plouc/nivo#2027] [Q&A] [answered]
I am trying to implement a bar chart where you can click bars to highlight them, and click the background to deselect them. Now, getting an event to fire when you click a bar is quite easy, as I can just add an onClick callback - I am, however, unsure as to how I could make an event fire when clicking on the background. Does anyone happen to have an idea of how to implement this?

**Comment by tkonopka** [comment, marked as answer]
Hi @noambugot1. You could create a custom layer with a plain rectangle, position it underneath the bars and grid and other layers, and process clicks on that. 

An example with a custom a layer in the background is [here](https://nivo.rocks/storybook/?path=/story/scatterplot--adding-extra-layers) with code [here](https://github.com/plouc/nivo/blob/master/packages/scatterplot/stories/ScatterPlot.stories.tsx#L558-L599). The example uses ScatterPlot, but the approach would be similar for Bar. Note the layers for Bar chart are `['grid', 'axes', 'bars', 'markers', 'legends', 'annotations']`.

The props passed to the custom layer contain the dimension of the chart and other things, but do not hold information about what has been clicked previosly. Hope this helps.
